crates/cli/src/opts/build/zksync.rs

```
@@ -85,7 +85,11 @@ pub struct ZkSyncArgs {
     /// Detect missing libraries, instead of erroring
     ///
     /// Currently unused
-    #[clap(long = "zk-detect-missing-libraries")]
+    #[clap(
+        long = "zk-detect-missing-libraries-deprecated",
```

Maybe we should just remove from here instead of deprecating and then use the option in build args to create the right zk project.

---

This is an intentional hack I had to do to show the warning when users currently have this flag in the TOML file.
If we remove all track of the flag for the rest of the commands (besides `build`), we can't look for it in the code.

If you have an alternative for this without creating a new flag argument, please let me know 🙏

---

Imo we should not warn about this because the usage in toml is rare and probably unintended. No working project will have this in config because it will yield no bytecode. The behaviour will no be that this silently does nothing and then if there are missing libraries it will fail compilation, which should make users aware of the issue (and then they can run the thing with the cli option)

---

You are right. I wanted to be extra cautious here, but the probability of someone having this running on a CI or something similar right now is very low. Let's go for the straigtforward solution